### PR TITLE
London | 26-ITP-May | Raihan Sharif | Sprint 3 | Implement and rewrite

### PR DESCRIPTION
**Pull request by RaihanSharif**
## Learners, PR Template

Self checklist

- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist

Completed as per spec.

### CODE REVIEW

File: Sprint-3/1-implement-and-rewrite-tests/implement/3-get-card-value.js [hunk 1]
```diff
@@ -22,7 +22,38 @@
...
+    "9",
+    "10",
+    "J",
+    "Q",
+    "K",
+  ];
+  const suit = ["♠", "♥", "♦", "♣"];
+
+  const cardValue = card.slice(0, -1);
+  const cardSuit = card[card.length - 1];
```
Comment (cjyuan) [inline Sprint-3/1-implement-and-rewrite-tests/implement/3-get-card-value.js:43]: `card,slice(-1)` also works.

File: Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/1-get-angle-type.test.js [hunk 2]
```diff
@@ -11,6 +11,34 @@ test(`should return "Acute angle" when (0 < angle < 90)`, () => {
...
+  expect(getAngleType(179)).not.toEqual("Straight angle");
+  expect(getAngleType(180)).toEqual("Straight angle");
+  expect(getAngleType(181)).not.toEqual("Straight angle");
+});
+test(`should return "Reflex angle" when (180 < angle < 361)`, () => {
+  // Test various acute angles, including boundary cases
+  expect(getAngleType(179)).not.toEqual("Reflex angle");
+  expect(getAngleType(371)).not.toEqual("Reflex angle");
+  expect(getAngleType(200)).toEqual("Reflex angle");
 });
```
Comment (cjyuan) [inline Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/1-get-angle-type.test.js]: - 360 is not a reflex angle.

- Why not introduce a category to test invalid angles, especially the boundary cases?

File: Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/2-is-proper-fraction.test.js [hunk 3]
```diff
@@ -8,3 +8,42 @@ const isProperFraction = require("../implement/2-is-proper-fraction");
...
+  expect(isProperFraction(2, 3)).toEqual(true);
+});
+
+test(`should return true when numerator and denominator both negative and numerator < denominator`, () => {
+  expect(isProperFraction(-3, -2)).toEqual(true);
+});
+
+test(`should return true when numerator is negative and numerator < denominator`, () => {
+  expect(isProperFraction(-3, 2)).toEqual(true);
+});
```
Comment (cjyuan) [inline Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/2-is-proper-fraction.test.js]: - I don't think your function can pass these tests?

- You can use pseudo-code and notations like `abs(...)` or `| ... |` in the descriptions to more 
concisely describe the conditions (the "when" part).

File: Sprint-3/1-implement-and-rewrite-tests/implement/2-is-proper-fraction.js [hunk 4]
```diff
@@ -31,3 +32,8 @@ function assertEquals(actualOutput, targetOutput) {
 
 // Example: 1/2 is a proper fraction
 assertEquals(isProperFraction(1, 2), true);
+assertEquals(isProperFraction(2, 1), false);
+assertEquals(isProperFraction(0, 0), false);
+assertEquals(isProperFraction(-0, 0), false);
+assertEquals(isProperFraction(-5, 4), true);
+assertEquals(isProperFraction(-4, -5), false);
```
Comment (cjyuan) [inline Sprint-3/1-implement-and-rewrite-tests/implement/2-is-proper-fraction.js]: Can your function pass these tests?

**Comment by RaihanSharif** [comment]
Rewrote all the code as per review.

**Comment by cjyuan** [comment]
Changes look good.

**Comment by RaihanSharif** [comment]
Thank you for your reviews, and for your patience.

**Comment by illicitonion** [comment]
Closing PR because the May ITP run has finished. Feel free to re-open if you're still working on it.
